File: app/src/main/java/com/example/dev/hazikura/fragment/Household/OutputFragment.java
package com.example.dev.hazikura.fragment.Household;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.example.dev.hazikura.R;

/**
 * Created by yokoro
 **/

public class OutputFragment extends Fragment {

    private MyListener2 myListener2;
    private TextView date;
    private EditText content;
    private EditText number;
    private EditText amount;

    public static OutputFragment newInstance() {
        OutputFragment fragment = new OutputFragment();
        return fragment;
    }

    @Override
    public void onCreate (Bundle saveInstanceState){
        super.onCreate(saveInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_output, container, false);

    }

    @Override
    public void onViewCreated(View view, Bundle saveInstanceState){
        super.onViewCreated(view, saveInstanceState);

        view.findViewById(R.id.dateButton2).setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                if(myListener2 != null){
                    myListener2.onClickButton2();
                }
            }
        });
    }

    @Override
    public  void onStart(){
        super.onStart();

        findView();
        init();
        Button button = (Button)getActivity().findViewById(R.id.write_output);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                // DBに登録
                saveList();
            }
        });
    }

    public interface MyListener2 {
        void onClickButton2();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if(context instanceof MyListener2){
            myListener2 = (MyListener2) context;
        }
    }

    @Override
    public void onDetach(){
        super.onDetach();
        myListener2 = null;
    }

    private void findView(){
        date = (TextView) getActivity().findViewById(R.id.output_date);
        content = (EditText) getActivity().findViewById(R.id.output_content);
        number = (EditText) getActivity().findViewById(R.id.output_number);
        amount = (EditText) getActivity().findViewById(R.id.output_amount);
    }

    private void init(){
        content.setText("");
        number.setText("");
        amount.setText("");
    }

    private void saveList(){
        String strDate = date.getText().toString();
        String strContent = content.getText().toString();
        String strNumber = number.getText().toString();
        String strAmount = amount.getText().toString();

        if(strDate.equals("日付を選択してください") || strAmount.equals("") || strNumber.equals("") || strContent.equals("")) {
            Toast.makeText(getActivity(), "全ての箇所を入力してください", Toast.LENGTH_SHORT).show();
        } else {
            int iNumber = Integer.parseInt(strNumber);
            int iAmount = Integer.parseInt(strAmount);
            DBAdapter dbAdapter = new DBAdapter(getActivity());
            dbAdapter.openDB();
            dbAdapter.saveOutgo(strDate, strContent, iNumber, iAmount);
            dbAdapter.closeDB();

            init();
        }
    }
}

File: app/src/main/java/com/example/dev/hazikura/fragment/household/OutputFragment.kt
package com.example.dev.hazikura.fragment.household

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

import com.example.dev.hazikura.R

/**
 * Created by yokoro
 */

class OutputFragment : Fragment() {

    private var myListener2: MyListener2? = null
    private lateinit var date: TextView
    private lateinit var content: EditText
    private lateinit var number: EditText
    private lateinit var amount: EditText

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.fragment_output, container, false)

    }

    override fun onViewCreated(view: View, saveInstanceState: Bundle?) {
        super.onViewCreated(view, saveInstanceState)

        view.findViewById<View>(R.id.dateButton2).setOnClickListener {
            if (myListener2 != null) {
                myListener2!!.onClickButton2()
            }
        }
    }

    override fun onStart() {
        super.onStart()

        findView()
        init()
        val button = activity?.findViewById<View>(R.id.write_output) as Button
        button.setOnClickListener {
            // DBに登録
            saveList()
        }
    }

    interface MyListener2 {
        fun onClickButton2()
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is MyListener2) {
            myListener2 = context
        }
    }

    override fun onDetach() {
        super.onDetach()
        myListener2 = null
    }

    private fun findView() {
        date = activity?.findViewById<View>(R.id.output_date) as TextView
        content = activity?.findViewById<View>(R.id.output_content) as EditText
        number = activity?.findViewById<View>(R.id.output_number) as EditText
        amount = activity?.findViewById<View>(R.id.output_amount) as EditText
    }

    private fun init() {
        content.setText("")
        number.setText("")
        amount.setText("")
    }

    private fun saveList() {
        val strDate = date.text.toString()
        val strContent = content.text.toString()
        val strNumber = number.text.toString()
        val strAmount = amount.text.toString()

        if (strDate == "日付を選択してください" || strAmount == "" || strNumber == "" || strContent == "") {
            Toast.makeText(activity, "全ての箇所を入力してください", Toast.LENGTH_SHORT).show()
        } else {
            val iNumber = Integer.parseInt(strNumber)
            val iAmount = Integer.parseInt(strAmount)
            val dbAdapter = DBAdapter(activity!!)
            dbAdapter.openDB()
            dbAdapter.saveOutgo(strDate, strContent, iNumber, iAmount)
            dbAdapter.closeDB()

            init()
        }
    }

    companion object {

        fun newInstance(): OutputFragment {
            return OutputFragment()
        }
    }
}
